templates/dockerize.yml

```diff
@@ -0,0 +1,119 @@
+      echo "##vso[task.setvariable variable=version]$VERSION"
+    workingDirectory: $(Build.SourcesDirectory)
+    displayName: Extract microservice version from pom.xml file --> Task
+
+  - task: Docker@2
+    inputs:
+      containerRegistry: ${{ parameters.containerServiceConnection }}
+      repository: '${{ parameters.containerRepo }}/${{ parameters.imageName }}'
+      command: 'buildAndPush'
+      Dockerfile: '**/Dockerfile'
+      tags: '$(version)'
+    displayName: Dockerize ${{ parameters.imageName }} and push --> Task
+
+  - task: ado-discord-webhook@1
+    condition: always()
+    inputs:
+      channelId: ${{ parameters.discordChannel }}
+      webhookKey: ${{ parameters.discordKey }}
+      name: '$(System.TeamProject) - Docker'
+      avatar: 'https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fcdn3.iconfinder.com%2Fdata%2Ficons%2Flogos-and-brands-adobe%2F512%2F97_Docker-512.png&f=1&nofb=1'
+      messageType: 'content'
+      content: |
+        **__Dockerization Report__**
+        **Status:** $(Agent.JobStatus)
+        **Branch:** $(Build.SourceBranch)
+        **Trigger:** $(Build.Reason)
+        **Message:** $(Build.SourceVersionMessage)
+        **Image:** ${{ parameters.imageName }}:$(version)
+        **Container Repo:** <${{ parameters.containerUrl }}>
+        **Logs:** <$(System.CollectionUri)$(System.TeamProject)/_build/results?buildId=$(Build.BuildId)&view=logs>
+        **Repo:** <$(Build.Repository.Uri)/tree/$(Build.SourceBranchName)>
+    displayName: Notification --> Task
+
+- ${{ if eq(parameters.useDynamicVersion, false) }}:
+  - task: Docker@2
+    inputs:
+      containerRegistry: ${{ parameters.containerServiceConnection }}
+      repository: '${{ parameters.containerRepo }}/${{ parameters.imageName }}'
+      command: 'buildAndPush'
+      Dockerfile: '**/Dockerfile'
+      tags: ${{ parameters.version }}
+    displayName: Dockerize ${{ parameters.imageName }} and push --> Task
+  
+  - task: ado-discord-webhook@1
+    condition: always()
+    inputs:
+      channelId: ${{ parameters.discordChannel }}
+      webhookKey: ${{ parameters.discordKey }}
+      name: '$(System.TeamProject) - Docker'
+      avatar: 'https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fcdn3.iconfinder.com%2Fdata%2Ficons%2Flogos-and-brands-adobe%2F512%2F97_Docker-512.png&f=1&nofb=1'
+      messageType: 'content'
+      content: |
+        **__Dockerization Report__**
+        **Status:** $(Agent.JobStatus)
+        **Branch:** $(Build.SourceBranch)
+        **Trigger:** $(Build.Reason)
+        **Message:** $(Build.SourceVersionMessage)
+        **Image:** ${{ parameters.imageName }}:${{ parameters.version }}
+        **Container Repo:** <${{ parameters.containerUrl }}>
+        **Logs:** <$(System.CollectionUri)$(System.TeamProject)/_build/results?buildId=$(Build.BuildId)&view=logs>
+        **Repo:** <$(Build.Repository.Uri)/tree/$(Build.SourceBranchName)>
+    displayName: Notification --> Task
```

Comment:
couldn't we reduce this duplication somehow?

Reply:
I put the common elements of the message at the end.  The only element that had to be handled differently between the two blocks was the version.

Reply:
Okay, so that didn't work because apparently, I can't store multi-line strings as the default value for a string parameter...
I tried `|`, `>`, `>-`, and plain `''`.
I reverted to the original version for now.

Reply:
parameters are a little unique, though i'm still surprised it doesn't work.
but it should work fine with variables.
doesn't have to be addressed in this PR.

Reply:
Okay, I will circle back to this on a future PR.